autorag/nodes/queryexpansion/base.py

```diff
@@ -28,8 +28,8 @@ def wrapper(
         if func.__name__ == "pass_query_expansion":
             return func(queries=queries)
 
-        # set module parameters
-        llm_str = kwargs.pop("llm")
+        # set generator module for query expansion
+        generator_callable, generator_param = make_generator_callable_param(kwargs)
```

Comment:
In kwargs, there are non-generator module parameters like 'prompt'. 
It looks like you need to pop that prompt first.

Plus, you don't need to pop batch parameter no more.